src/main/docker/mysql.yml

```
@@ -1,19 +1,40 @@
-version: '2.4'
+# ----------------------------------------------------------------------------------------------------------------------
+# MySQL base service
+# ----------------------------------------------------------------------------------------------------------------------
+
 services:
-    artemis-mysql:
-        image: mysql:8.0.32
-        # volumes:
-        #     - ~/volumes/jhipster/Artemis/mysql/:/var/lib/mysql/
+    mysql:
+        container_name: artemis-mysql
+        image: docker.io/library/mysql:8.0.32
+        pull_policy: always
+        volumes:
+            - artemis-mysql-data:/var/lib/mysql
         environment:
-            - MYSQL_ALLOW_EMPTY_PASSWORD=yes
-            - MYSQL_ROOT_PASSWORD=
-            - MYSQL_DATABASE=Artemis
+            MYSQL_ALLOW_EMPTY_PASSWORD: "yes"
+            MYSQL_ROOT_PASSWORD:
+            MYSQL_DATABASE: "Artemis"
         ports:
-            - 3306:3306
+            - "3306:3306"
```

Since we're now using this also for production - we have to limit access to the port! 
```suggestion
            - "127.0.0.1:3306:3306"
```

---

Do you need the binding to localhost at all? Inside the container you can easily use the mysql command and within the artemis network, the artemis instance can access the exposed port without binding to localhost

---

@dfuchss Yes some developers are using the `mysql` container in combination with an Artemis Instance started from the IDE and running on localhost.

@Mtze If we really want to use this for production from now on, we should maybe add an override and then also not allow empty passwords, ...

Not sure how many have setups where they are connection to non localhost MySQL servers and localhost wouldn't be enough.

---

So - you designed the whole PR for dev only?

---

No. This PR was originally for refactoring and clean-up of the current docker compose setups. It's main purpose, I understood, was to get developers up and running easily. Afaik this didn't include a production MySQL. But it's easily possible to add one tomorrow with overrides, as I did for the Artemis prod and dev docker compose setups :wink: